### ISSUE TITLE
One click installation of extensions

### ISSUE THREAD

**Issue by owaincuvelier** [humanmade/backupwordpress#525]
It is possible to use the Stripe AP and EDD to enable one click installation of extensions from the WordPress dashboard. It was demonstrated by Thomas Griffin (can't find the link though), this would surely increase conversions from the free BackUpWordPress plugin.

I'm not sure how this would work with EDD though. But it's a cool idea

**Comment by pdewouters** [comment]
Would be nice to replace the upsell links with "install" buttons. We could mimick the plugins search results design and use the same ajax install process.clicking install would popup a modal, or slide-down a purchase form.
we may be able to see thie imlementation as an EDD addon.

**Comment by pdewouters** [comment]
So this is the flow I had in mind

- user visits addons catalog page
- User clicks "install"
- Display payment form
- On success install addon from remote location
- Successful payment should trigger an EDD entry and send license to user
- Install successful - prompt user for license key.
- finished

We would need to be able to use the Software Licensing addon API to trigger a new purchase if payment is successful

https://easydigitaldownloads.com/support/topic/create-purchase-via-api/

**Comment by willmot** [comment]
It would be nice if it also automatically entered the correct license, that way the process could be completely seamless / "it just works"

**Comment by pdewouters** [comment]
yes, it would. I suppose if the API returns the license on a successful request.

**Comment by pdewouters** [comment]
the EDD API doesn't support creating license keys, so we'd still need to send an email with the key for then to enter manually.

**Comment by pdewouters** [comment]
Relevant code example https://github.com/humanmade/hm-stream/blob/f77afa5515daf3b0aa8c0226343812816271e3ea/includes/extensions.php

**Comment by willmot** [comment]
The first bash at this is implemented in #909. We'll leave this open to cover future enhancements, including moving the purchasing flow inline.

**Comment by libby-barker** [comment]
@willmot and @pdewouters what work remains on this one?

**Comment by willmot** [comment]
We have an extensions page within the plugin. However clicking install on one of those extensions just links the user out to the bwp.hmn.md website where they then have to complete the normal purchase flow, receive an email with links to download the extension plugins, download those plugins, upload them to their site, activate them and then paste in the correct API key.

Instead we should work to bring that entire process inline as part of the plugin. The flow outlined in https://github.com/humanmade/backupwordpress/issues/525#issuecomment-81619336 is still relevant. Although I think we could also auto-populate the license key.
